docs/guides/watcher.md

```diff
@@ -0,0 +1,114 @@
+# Guide: Creating a Whale Watcher Agent
+
+This guide walks you through creating a whale watcher agent that monitors blockchain transactions for transfers.
+
+**Note**: Replace `author` in the commands below with your author name (e.g., alice, bob, etc.)
+
+## 1. Create the FSM Definition
+
+Create `whale_watcher_fsm.yaml`:
+
+```yaml
+alphabet_in:
+  - BLOCK_RECEIVED        # A new Ethereum block is detected
+  - TX_OVER_THRESHOLD     # A transaction exceeds the whale threshold
+  - TX_UNDER_THRESHOLD    # A transaction is under the whale threshold
+  - DONE                  # All transactions for this block are processed
+  - TIMEOUT              # A timeout or other error occurs
+
+default_start_state: SetupRound
+label: WhaleWatcherAbciApp
+
+states:
+  - SetupRound
+  - IdleRound
+  - BlockReceivedRound
+  - AlertRound
+  - DoneRound
+  - ErrorRound
+
+start_states:
+  - SetupRound
+
+final_states:
+  - ErrorRound
+  - DoneRound
+
+transition_func:
+  (SetupRound, DONE): IdleRound
+  (IdleRound, BLOCK_RECEIVED): BlockReceivedRound
+  (BlockReceivedRound, TX_OVER_THRESHOLD): AlertRound
+  (BlockReceivedRound, TX_UNDER_THRESHOLD): BlockReceivedRound
+  (BlockReceivedRound, DONE): DoneRound
+  (AlertRound, DONE): DoneRound
+  (DoneRound, DONE): SetupRound
+  (IdleRound, DONE): SetupRound
+  (BlockReceivedRound, TIMEOUT): ErrorRound
+  (AlertRound, TIMEOUT): ErrorRound
+```
+
+## 2. Create Agent from FSM
+
+Generate the initial agent structure from the FSM definition:
+
+```bash
+adev create_from_fsm author/whale_watcher whale_watcher_fsm.yaml
```

Comment:
Does this not just work?

Reply:
yeah but this doesn't have the business logic ideally want to explain how to do that also